lib/vdc/BOVCollection.cpp

```diff
@@ -0,0 +1,368 @@
+
+        if (ss.eof() == 0) {
+            std::string message = "The keyword " + token + " may only contain one value.";
+            SetErrMsg(message.c_str());
+            return -1;    // Bad token
+        }
+
+        if (ss.bad()) {
+            std::string message = "Invalid value for " + token + " in BOV header file.";
+            SetErrMsg(message.c_str());
+            return -1;    // Bad token
+        }
+        return 1;    // we now have a token
+    }
+    return 0;    // we didn't find the token
+}
+
+// Template specialization for reading data of type std::vector<int> or std::vector<float>
+// All std::vectors returned must have a size of 3
+template<typename T> int BOVCollection::_findToken(const std::string &token, std::string &line, std::vector<T> &value, bool verbose)
+{
+    // Skip comments
+    if (line[0] == '#') { return 0; }
+
+    size_t pos = line.find(token);
+    if (pos != std::string::npos) {    // We found the token
+        T lineValue;
+        _findTokenValue(line);
+        std::stringstream lineStream(line);
+
+        value.clear();
+        while (lineStream >> lineValue) { value.push_back(lineValue); }
+
+        if (value.size() != 3) {
+            value.clear();
+            std::string message = token + " must be a set of three values.";
+            SetErrMsg(message.c_str());
+            return -1;    // Bad token
+        }
+
+        if (lineStream.bad()) {
+            value.clear();
+            std::cout << "FAIL v " << token << std::endl;
+            std::string message = "Invalid value for " + token + " in BOV header file.";
+            SetErrMsg(message.c_str());
+            return -1;    // Bad token
+        }
+
+        if (verbose) {
+            std::cout << std::setw(20) << token << " ";
+            for (int i = 0; i < value.size(); i++) std::cout << value[i] << " ";
+            std::cout << std::endl;
+        }
+        return 1;    // we now have a token
+    }
+    return 0;    // we didn't find the token
+}
+
+void BOVCollection::_findTokenValue(std::string &line) const
+{
+    std::string delimiter = ": ";
+
+    size_t      pos = 0;
+    std::string token;
+    while ((pos = line.find(delimiter)) != std::string::npos) {
+        token = line.substr(0, pos);
+        line.erase(0, pos + delimiter.length());
+    }
+}
+
+size_t BOVCollection::_sizeOfFormat(DC::XType type) const
+{
+    switch (type) {
+    case DC::XType::INT32: return 4;
+    case DC::XType::FLOAT: return 4;
+    case DC::XType::DOUBLE: return 8;
+    default: return -1;
+    }
+}
+
+void BOVCollection::_swapBytes(void *vptr, size_t size, size_t n) const
+{
+    unsigned char *ucptr = (unsigned char *)vptr;
+    unsigned char  uc;
+    size_t         i, j;
+
+    for (j = 0; j < n; j++) {
+        for (i = 0; i < size / 2; i++) {
+            uc = ucptr[i];
+            ucptr[i] = ucptr[size - i - 1];
+            ucptr[size - i - 1] = uc;
+        }
+        ucptr += size;
+    }
+}
```

Comment:
Did you test that this function really change between big endian and small endian? (it doesn't look so for me.)

My suggestion is to not support the endian token at all, since it's actually pretty difficult to switch between the two.

Reply:
Never mind, I got a wrong understanding of big/little endian.